### PR TITLE
Fixed - can not directly access anonymous API with API URL

### PR DESCRIPTION
**Pull request by shnrndk**
Fixed redirection to dev portal if the user tries to access an anonymous API using a direct URL
Fix #https://github.com/wso2/api-manager/issues/23

### CODE REVIEW

**Review by chanaka3d — APPROVED** [review]

**Review by chanaka3d — APPROVED** [review]

File: portals/devportal/src/main/webapp/services/login/login_callback.jsp [hunk 1]
```diff
@@ -84,8 +84,20 @@
         postLogoutRedirectURI = serverUrl + appContext + LOGOUT_CALLBACK_URL_SUFFIX;
     }
     String error = request.getParameter("error");
-    if (error != null && error.equals("login_required")) {
-        response.sendRedirect(postLogoutRedirectURI + "?referrer=" + referrer);
+    boolean isAnonymousEnabled = false;
+    if ((error != null) && error.equals("login_required")) {
+        if(state!=null){
```
Comment (shilmyhasan) [inline portals/devportal/src/main/webapp/services/login/login_callback.jsp:89]: Formatting error, please check other places as well.

File: portals/devportal/src/main/webapp/services/login/login_callback.jsp [hunk 2]
```diff
@@ -84,8 +84,20 @@
...
+            state = URLDecoder.decode(state, "UTF-8");
+            if(state.contains("/apis") == true || state.contains("/home") == true){
```
Comment (lasanthaS) [inline portals/devportal/src/main/webapp/services/login/login_callback.jsp:91]: Could be simply rewritten as follows.

```javascript
if (state.contains("/apis") || state.contains("/home")) {
```

Comment (shnrndk) [reply portals/devportal/src/main/webapp/services/login/login_callback.jsp:91]: Fixed in https://github.com/wso2/apim-apps/pull/523

File: portals/devportal/src/main/webapp/services/login/login_callback.jsp (hunk 1 above)
Comment (shnrndk) [reply portals/devportal/src/main/webapp/services/login/login_callback.jsp:89]: Fixed in https://github.com/wso2/apim-apps/pull/523

**Pull request merged**